plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -515,39 +532,37 @@ fu_mediatek_scaler_device_get_staged_data(FuMediatekScalerDevice *self,
 
 static gboolean
 fu_mediatek_scaler_device_check_sent_info(FuMediatekScalerDevice *self,
-					  GInputStream *stream,
+					  FuChunk *chk,
+					  guint32 sent_size,
 					  GError **error)
 {
 	guint16 chksum = 0;
 	guint16 sum16 = 0;
 	guint32 pktcnt = 0;
-	gsize streamsz = 0;
 
-	if (!fu_mediatek_scaler_device_get_staged_data(self, &chksum, &pktcnt, error))
+	if (!fu_mediatek_scaler_device_get_staged_data(self, &chksum, &pktcnt, error)) {
+		g_prefix_error(error, "failed to get the staged data: ");
 		return FALSE;
+	}
 
 	/* verify the staged packets on chip */
-	if (!fu_input_stream_size(stream, &streamsz, error))
-		return FALSE;
-	if (streamsz != pktcnt) {
+	if (sent_size != pktcnt) {
 		g_set_error(error,
 			    FWUPD_ERROR,
-			    FWUPD_ERROR_WRITE,
-			    "failed data verification, sent size: %" G_GSIZE_FORMAT
-			    ", ack size: %u",
-			    streamsz,
+			    FWUPD_ERROR_INVALID_DATA,
+			    "data packet size mismatched, expected: %X, chip got: %X",
+			    sent_size,
 			    pktcnt);
 		return FALSE;
 	}
 
 	/* verify the checksum on chip */
-	if (!fu_input_stream_compute_sum16(stream, &sum16, error))
-		return FALSE;
+	sum16 = fu_sum16(fu_chunk_get_data(chk), fu_chunk_get_data_sz(chk));
```

Comment:
still not sure about this; why not use `fu_input_stream_compute_sum16`?

Reply:
we've sliced the stream into chunks, wondering how to use `fu_input_stream_compute_sum16` here?

Reply:
Aha, I missed the chunking -- ignore me. Thanks.